### ISSUE TITLE
Got error when I run play-scala-starter-example project

### ISSUE THREAD

**Issue by Siceberg** [playframework/playframework#7704]
I'm a novice of Scala and play framework, When I try 'sbt run' and visit the play-scala-starter-example project, I got error messages:

```
sbt run
[warn] Executing in batch mode.
[warn]   For better performance, hit [ENTER] to switch to interactive mode, or
[warn]   consider launching sbt without any commands, or explicitly passing 'shell'
[info] Loading project definition from /Users/leo/Developer/Scala/play-scala-starter-example/project
[info] Updating {file:/Users/leo/Developer/Scala/play-scala-starter-example/project/}play-scala-starter-example-build...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
[info] Set current project to play-scala-starter-example (in build file:/Users/leo/Developer/Scala/play-scala-starter-example/)
[info] Updating {file:/Users/leo/Developer/Scala/play-scala-starter-example/}root...
[info] Resolving jline#jline;2.14.3 ...
[info] Done updating.

--- (Running the application, auto-reloading is enabled) ---

[info] p.c.s.AkkaHttpServer - Listening for HTTP on /0:0:0:0:0:0:0:0:9000

(Server started, use Enter to stop and go back to the console...)

[info] Compiling 14 Scala sources and 1 Java source to /Users/leo/Developer/Scala/play-scala-starter-example/target/scala-2.12/classes...
[error] (compile:compileIncremental) javac returned nonzero exit code
[info] Compiling 14 Scala sources and 1 Java source to /Users/leo/Developer/Scala/play-scala-starter-example/target/scala-2.12/classes...
[error] (compile:compileIncremental) javac returned nonzero exit code
[error] application - 

! @750boi5mo - Internal server error, for (GET) [/] ->
 
play.sbt.PlayExceptions$UnexpectedException: Unexpected exception[The compilation failed without reporting any problem!]
	at play.sbt.run.PlayReload$$anonfun$taskFailureHandler$1$$anonfun$apply$9.apply(PlayReload.scala:50)
	at play.sbt.run.PlayReload$$anonfun$taskFailureHandler$1$$anonfun$apply$9.apply(PlayReload.scala:50)
	at scala.Option.getOrElse(Option.scala:120)
	at play.sbt.run.PlayReload$$anonfun$taskFailureHandler$1.apply(PlayReload.scala:50)
	at play.sbt.run.PlayReload$$anonfun$taskFailureHandler$1.apply(PlayReload.scala:44)
	at scala.Option.map(Option.scala:145)
	at play.sbt.run.PlayReload$.taskFailureHandler(PlayReload.scala:44)
	at play.sbt.run.PlayReload$.compileFailure(PlayReload.scala:40)
	at play.sbt.run.PlayReload$$anonfun$compile$1.apply(PlayReload.scala:17)
	at play.sbt.run.PlayReload$$anonfun$compile$1.apply(PlayReload.scala:17)
Caused by: sbt.compiler.CompileFailed: null
	at sbt.compiler.javac.JavaCompilerAdapter.compileWithReporter(JavaCompilerAdapter.scala:35)
	at sbt.compiler.javac.AnalyzingJavaCompiler$$anonfun$compile$1.apply$mcV$sp(AnalyzingJavaCompiler.scala:65)
	at sbt.compiler.javac.AnalyzingJavaCompiler$$anonfun$compile$1.apply(AnalyzingJavaCompiler.scala:65)
	at sbt.compiler.javac.AnalyzingJavaCompiler$$anonfun$compile$1.apply(AnalyzingJavaCompiler.scala:65)
	at sbt.compiler.javac.AnalyzingJavaCompiler.timed(AnalyzingJavaCompiler.scala:93)
	at sbt.compiler.javac.AnalyzingJavaCompiler.compile(AnalyzingJavaCompiler.scala:64)
	at sbt.compiler.MixedAnalyzingCompiler$$anonfun$compileJava$1$1.apply$mcV$sp(MixedAnalyzingCompiler.scala:60)
	at sbt.compiler.MixedAnalyzingCompiler$$anonfun$compileJava$1$1.apply(MixedAnalyzingCompiler.scala:60)
	at sbt.compiler.MixedAnalyzingCompiler$$anonfun$compileJava$1$1.apply(MixedAnalyzingCompiler.scala:60)
	at sbt.compiler.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:74)
[info] Compiling 14 Scala sources and 1 Java source to /Users/leo/Developer/Scala/play-scala-starter-example/target/scala-2.12/classes...
[error] (compile:compileIncremental) javac returned nonzero exit code
[info] Compiling 14 Scala sources and 1 Java source to /Users/leo/Developer/Scala/play-scala-starter-example/target/scala-2.12/classes...
[error] (compile:compileIncremental) javac returned nonzero exit code
[INFO] [08/07/2017 15:26:45.087] [Thread-3] [CoordinatedShutdown(akka://play-dev-mode)] Starting coordinated shutdown from JVM shutdown hook
```

I tried command `sbt new playframework/play-scala-seed.g8` to build and run a new project, I got the same messages. I don't know what to do next, hope someone can help me, thanks.

### Play Version

I find the commands from Play 2.6.2 Document

### Operating System

MacOS 10.12.6

### JDK

java version "1.8.0_92"
Java(TM) SE Runtime Environment (build 1.8.0_92-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.92-b14, mixed mode)

**Comment by marcospereira** [comment]
This is weird, especially this line:

```
play.sbt.PlayExceptions$UnexpectedException: Unexpected exception[The compilation failed without reporting any problem!]
```
Could you please try with another version of the JDK? I have no problems with the following Java version:

```
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)
```

**Comment by richdougherty** [comment]
@Siceberg, I saw you typed `sbt run` from the command line. Does the same thing happen if you type `sbt` from the command line then `run` once you see sbt's `>` prompt?

**Comment by richdougherty** [comment]
Also would you mind including a full script, starting from `sbt new playframework/play-scala-seed.g8` and including anytime you `cd` into other directories?

**Comment by Siceberg** [comment]
Hi @marcospereira @richdougherty , thank you for your reply.

I checked the Java Control Panel of my Mac, it shows java version is 1.8.0_144, which is inconsistent with the command line results. I think this is the result of my upgrade using the Java Control Panel. After that I downloaded jdk-8u144 complete installation package and reinstalled it,  then everything worked fine.

Does this mean that using the Java Control panel upgrade is dangerous?
